### DISCUSSION TITLE
[BOUNTY]: I'm paying for a fake android assistant app that just opens the URL and starts recording

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#4734] [General]
Hi,

I've been using [SpeakGPT](https://github.com/AndraxDev/speak-gpt) for a while but now that I'm moving to self hosted stuff the only really lacking feature for me is an android assistant feature.

Currently my setup is to have brave browser on my android phone, and used the "add to homescreen" feature to create a PWA that goes straight to my open webui instance.

What I'm after is a simple android app that registers as an assistant but when toggled opens my openwebui instance's URL and starts recording a new message.

If possible using my brave PWA, or using my default browser (brave), or as a "standalone browser" of some sort (I think it's called webview?). Note that I have HTTPs setup on my instance so recording permission shouldn't be an issue. Also if possible without having to recompile the apk if I change my URL or port.

Of course if anyone knows for a fact that it's not possible, thanks for telling me :)

Being unable to code in kotlin I can't do this simple app myself so I thought it could be a nice fit for a bounty.

Is anyone interested? The price of course depends on what's considered fair by the person who is down for it.

Edit: if it's possible via tasker that can be fine with me!

**Comment by thiswillbeyourgithub** [comment]
Update: i think i found a way. There's a seemingly great app called [hermit](https://hermit.chimbori.com/) that allows creating PWA. Their premium features allows injecting custom scripts so triggering the microphone should be doable. For the gapps less android users they allow purchasing the apk on their own website. Also they have a guide on how to create an actual apk there https://hermit.chimbori.com/help/apk so maybe adding assistant permission is trivial.

I'll probably try this at the end of the summer.

**Reply by 1661410107114145152525** [reply]
Any news here @thiswillbeyourgithub ? Thanks for the great tipp also! Do you think this also helps when using selfhosted Immich with the browser version instead? E.g. speed up loading times and so on? How does it compare with Chromes/Braves "install" feature (it also adds to homescreen but strips out adress bar etc.) And am I able to e.g. run it as sth. that hides the navbar etc. or have bettet caching so images are cached longer / better?

**Reply by thiswillbeyourgithub** [reply]
Hi. I have switched phone and out of laziness just used brave > install instead of hermit. So I have not progressed on ways to use it as an android assistant directly.

I have not noticed performance differences between using hermit versus brave.

I think that "install" from brave forces you to a specific URI whereas with hermit you can specify it yourself. That means that you could use it to go directly to a specific album or something in immich.

I think hermit has some privacy settings that might be good but not sure how good they are compared to the ones of brave for instance.

I recommend giving hermit a try to make your own opinon.

I'm sure there are way to vibecode an assistant shortcut that opens open-webui directly but won't have the time.

**Reply by 1661410107114145152525** [reply]
@thiswillbeyourgithub Thanks for your quick reply and the time! :)
